libnavigation-base/src/test/java/com/mapbox/navigation/base/route/NavigationRouteEqualsTest.kt

```diff
@@ -0,0 +1,184 @@
+                    false
+                ),
+                arrayOf(
+                    "different waypoints from the same source",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints1FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints2FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    false
+                ),
+                arrayOf(
+                    "different waypoints from different sources",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints1FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    createDirectionsResponse(
+                        routes = listOf(route1WithWaypoints1FromResponse),
+                        unrecognizedProperties = unrecognizedWaypoints2
+                    ),
+                    createRouteOptions(waypointsPerRoute = false),
+                    "id#0",
+                    false
+                ),
+                arrayOf(
+                    "same waypoints from the same source",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints1FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints1FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    true
+                ),
+                arrayOf(
+                    "same waypoints from different source",
+                    createDirectionsResponse(routes = listOf(route1WithWaypoints1FromRoute)),
+                    createRouteOptions(waypointsPerRoute = true),
+                    "id#0",
+                    createDirectionsResponse(
+                        routes = listOf(route1WithWaypoints1FromResponse),
+                        unrecognizedProperties = unrecognizedWaypoints1
+                    ),
+                    createRouteOptions(waypointsPerRoute = false),
+                    "id#0",
+                    false
+                ),
+            )
+        }
+    }
+
+    private lateinit var route1: NavigationRoute
+    private lateinit var route2: NavigationRoute
+
+    @Before
+    fun setUp() {
+        route1 = NavigationRoute(
+            directionsResponse1,
+            0,
+            routeOptions1,
+            mockk(relaxUnitFun = true) {
+                every { routeId } returns id1
+                every { responseUuid } returns "uuid#0"
+                every { routerOrigin } returns RouterOrigin.ONBOARD
+                every { routeInfo } returns RouteInfo(listOf(mockk(relaxed = true)))
+            }
+        )
```

Comment:
maybe? 
```suggestion
        route1 = createNavigationRoutes(
            response = directionsResponse1,
            options = routeOptions1
        ).first()
```

Reply:
I need to mock id. I can use spy but IMO switching to `createNavigationRoutes` gives no profit here.

Reply:
`createNavigationRoutes` generates in the same way as NN does, so it's very similar, if not identical, to how the real system works. is that enough for unit tests?

Reply:
How can I mock the id if I use `createNavigationRoutes`?
I only see they way with creating a spy.
My question is: why is it better than the current approach?